api/test/wfl/integration/firecloud_test.clj

```diff
@@ -31,15 +31,14 @@
   (util/bracket
    #(firecloud/create-submission workspace method-configuration entity)
    #(firecloud/abort-submission workspace %)
-   #(let [workflow (-> (firecloud/get-submission workspace %) :workflows first)]
+   #(let [get-workflows (fn [] (:workflows (firecloud/get-submission workspace %)))
+          workflow      (first (get-workflows))]
       (is (= (second entity) (get-in workflow [:workflowEntity :entityName])))
       (is (#{"Queued" "Submitted"}
            ;; GH-1212: `get-submission` can return the workflow before it has
            ;; been queued. In such cases :status is nil so try again
-           (or (:status workflow) (-> (firecloud/get-submission workspace %)
-                                      :workflows
-                                      first
-                                      :status)))))))
+           (or (:status workflow)
+               (util/poll (comp :status first get-workflows))))))))
```

Comment:
(Commenting only to confirm my own understanding)

- We pass a function that is itself a composition of functions to `poll` so that `poll` can call it anew for each round.
- If the passed-in function evaluates to false, `poll` will keep trying.  A `nil` status evaluates to boolean false.

Reply:
Exactly